### PR TITLE
Substantially (~2x) speedup iteration time for folks working on LSP.

### PR DESCRIPTION
**Pull request by DarkDimius**
Do not link against LSP, Autogen, Proto & metrics in sorbet-orig.

### Motivation
Faster iteration time.
I've tried to do it using smart macros similar to `DEBUG_ONLY` that we already have, but IMHO it looked worse.

While I hate the implementation of this PR, I think it's worth it as it speeds up folks who work outside of Sorbet core.

### Test plan

See included automated tests.

### CODE REVIEW

File: main/realmain.cc [hunk 1]
```diff
@@ -1,22 +1,28 @@
-// has to be included first as it violates our poisons
-#include "core/proto/proto.h"
-
+#ifdef SORBET_REALMAIN_MIN
+// minimal build to speedup compilation. Remove extra features
+#define FULL_BUILD_ONLY(X) false; // do nothing
```
Comment (jez) [inline main/realmain.cc]: Is this macro unused?

File: main/realmain.cc [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
+#else
+#define FULL_BUILD_ONLY(X) X;
```
Comment (jez) [inline main/realmain.cc]: If this is not unused, should the definition be like this?

```suggestion
#define FULL_BUILD_ONLY(X) (X);
```

**Comment by jez** [comment]
Not 100% clear to me why this speeds up the build, but I'm happy to try it out and report problems if they arise.

**Comment by DarkDimius** [comment]
> why this speeds up the build

Sorbet-orig will no longer even compile LSP. Which means that if someone is working on LSP, their changes will not invalidate `sorbet-orig` and thus won't invalidate everything downstream including payload.

**Pull request merged**